Aj okta729831 placeholder 8 (#4894)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/concepts/scim/index.md

This topic covers the concepts and use cases for SCIM and Okta.

When you’re ready to create a SCIM integration, see [Build a SCIM provisioning integration](/docs/guides/scim-provisioning-integration-overview) and [how to implement the SCIM protocol with Okta](/docs/reference/scim/).

## What is SCIM for?

SCIM, or the [System for Cross-domain Identity Management](https://scim.cloud) specification, is an open standard designed to manage user identity information. SCIM provides a defined schema for representing users and groups, and a RESTful API to run CRUD operations on those user and group resources.

The goal of SCIM is to securely automate the exchange of user identity data between your company's cloud apps and any Service Providers, such as enterprise SaaS apps.

> Managing user lifecycles in your org is a fundamental business problem. Hiring employees is just the first step. You also need to provision apps needed for their job, enforce corporate security policies, and update user accounts as they advance through your company. At the end of their employment, you need to ensure that all access is quickly and thoroughly revoked across all apps. Handling all of this can be time-consuming and error-prone if done manually.

Reviewer: ```suggestion
Managing user lifecycles in your org is a fundamental business problem. Hiring employees is just the first step. You also need to provision apps needed for their job and enforce corporate security policies. Then you need to update user accounts as they advance through your company. At the end of their employment, you need to ensure that all access is quickly and thoroughly revoked across all apps. Handling all of this can be time-consuming and error-prone if done manually.
```

As your company grows, the number of user accounts and provisioned software apps increases. Requests to add and remove users, reset passwords, change permissions, and add new types of accounts all take up valuable IT department time.

With the SCIM protocol, user data is stored in a consistent way and can be shared with different apps. Since data is transferred automatically, complex exchanges are simplified and the risk of error is reduced.

Adopting SCIM for domain management improves overall security for your company. Employees no longer need to sign in to each of their accounts. As teams develop new workflows and adopt new apps, your company can ensure security policy compliance across all accounts.

## How does Okta help?

[Okta Lifecycle Management](https://help.okta.com/okta_help.htm?id=ext_Provisioning_Deprovisioning_Overview) is a platform solution to provision and manage user accounts in cloud-based apps. Okta serves as a universal directory for identity-related information, giving the following benefits:

* IT departments can manage the user provisioning lifecycle through a single system.
* New employees are automatically provisioned with a user account for their apps.

> * Employee accounts can be created either directly from Okta accounts, or shared from external systems like HR apps or Active Directory.

Reviewer: ```suggestion
* You can create employee accounts either directly from Okta accounts or shared from external systems like HR apps or Active Directory.
```

* Any profile updates - like department changes - populate automatically.
* Inactive employees are automatically deactivated from their apps.

> > **Note**: Okta event hooks provide a mechanism for outbound calls to notify your own systems of events occurring in your Okta org and allow them to take actions as a result — for example, provisioning external app access after a user account is created in Okta. As such, they’re a flexible and lightweight alternative to SCIM. See [Event hook implementation](/docs/guides/event-hook-implementation/nodejs/main/).

Reviewer: ```suggestion
> **Note**: Okta event hooks provide a mechanism for outbound calls to notify your own systems of events occurring in your Okta org. This allows them to take actions as a result, for example, provisioning external app access after a user account is created in Okta. Event hooks are a flexible and lightweight alternative to SCIM. See [Event hook implementation](/docs/guides/event-hook-implementation/nodejs/main/).
```

The provisioning actions performed by an integration are described using the database operation acronym "CRUD": Create, Read, Update, and Delete. The four CRUD operations are the building blocks that combine to solve your end-to-end use cases:

### Create

This SCIM operation creates users in your downstream app based on the values in the Okta user profile and group assignments.

### Read

Information about user and group resources can be queried from your app to match them against existing Okta resources. If the resources don't exist, they can be imported into Okta.

### Update

If a resource in your app needs to be updated based on data changed in Okta, this operation updates existing user or group attributes. Alternatively, if your app functions as the source of truth for specific attributes of a user identity, this action updates the Okta user profile.

User attributes can be mapped from your source into Okta. Conversely, an attribute can be mapped from Okta to a target attribute in your app.

### Delete (Deprovision)

Deleting or deprovisioning user profiles in SCIM operations depends on whether Okta or your SCIM app is the source of truth for user profile information.

> * If an admin deprovisions an user's profile inside Okta, the user resource inside your SCIM app is updated with `active=false`. If that user needs to be reprovisioned later (for example, a return from parental leave or if a contractor is rehired), then the `active` attribute can be switched back to `true`.

Reviewer: ```suggestion
* If an admin deprovisions a user's profile inside Okta, the user resource inside your SCIM app is updated with `active=false`. If that user needs to be reprovisioned later (for example, a return from parental leave or if a contractor is rehired), then you can switch back the `active` attribute to `true`.
```

* Similarly, if a user profile is deleted from inside your SCIM app, and that SCIM app sources the user, then when an import from your SCIM app is run, the user's profile is deleted inside Okta.

### Sync passwords

Outside of the base CRUD operations, Okta supports other provisioning features like syncing passwords.

Password synchronization helps you coordinate Okta-sourced users to ensure that a user’s Active Directory (AD) password and their Okta password always match. With password synchronization, your users have a single password to access apps and devices.

This option sets the user's password for your integration to match the Okta password or to be assigned a randomly generated password. For more information about this functionality and how to configure it in the Okta product, see [Synchronize passwords from Okta to Active Directory](https://help.okta.com/okta_help.htm?id=ext_Security_Using_Sync_Password).

### Map profile attributes

Another provisioning feature supported by Okta is the mapping of user profile attributes.

> After enabling provisioning, you can set an app as the "source" from which users profiles are imported or the "target" to which attributes are sent.

Reviewer: ```suggestion
After enabling provisioning, you can set an app as the "source" from which user profiles are imported or the "target" to which attributes are sent.
```
